src/ovs/aca_ovs_programmer.cpp

```diff
@@ -34,51 +39,78 @@ ACA_OVS_Programmer &ACA_OVS_Programmer::get_instance()
+  setup_ovs_bridges_mutex.lock();
+
+  // check to see if br-int and br-tun is already there
+  execute_ovsdb_command("br-exists br-int", not_care_culminative_time, overall_rc);
+  bool br_int_existed = (overall_rc == EXIT_SUCCESS);
+
+  execute_ovsdb_command("br-exists br-tun", not_care_culminative_time, overall_rc);
+  bool br_tun_existed = (overall_rc == EXIT_SUCCESS);
+
+  overall_rc = EXIT_SUCCESS;
+
+  // case 1: both br-int and br-tun exist
+  if (br_int_existed && br_int_existed) {
+    // nothing to do
+  }
+
+  // case 2: both br-int and br-tun not there, create them
+  else if (!br_int_existed && !br_int_existed) {
+    // create br-int and br-tun bridges
+    execute_ovsdb_command("add-br br-int", not_care_culminative_time, overall_rc);
 
-  // TODO: see if br-int and br-tun is already there, one option is the delete
-  // them and start everything from scratch
+    execute_ovsdb_command("add-br br-tun", not_care_culminative_time, overall_rc);
 
-  // create br-int and br-tun bridges
-  execute_ovsdb_command("add-br br-int", not_care_culminative_time, overall_rc);
+    // execute_openflow_command("del-flows br-tun", not_care_culminative_time, overall_rc);
 
-  execute_ovsdb_command("add-br br-tun", not_care_culminative_time, overall_rc);
+    // create and connect the patch ports between br-int and br-tun
+    execute_ovsdb_command("add-port br-int patch-tun", not_care_culminative_time, overall_rc);
 
-  execute_openflow_command("del-flows br-tun", not_care_culminative_time, overall_rc);
+    execute_ovsdb_command("set interface patch-tun type=patch",
+                          not_care_culminative_time, overall_rc);
 
-  // create and connect the patch ports between br-int and br-tun
-  execute_ovsdb_command("add-port br-int patch-tun", not_care_culminative_time, overall_rc);
+    execute_ovsdb_command("set interface patch-tun options:peer=patch-int",
+                          not_care_culminative_time, overall_rc);
 
-  execute_ovsdb_command("set interface patch-tun type=patch",
-                        not_care_culminative_time, overall_rc);
+    execute_ovsdb_command("add-port br-tun patch-int", not_care_culminative_time, overall_rc);
 
-  execute_ovsdb_command("set interface patch-tun options:peer=patch-int",
-                        not_care_culminative_time, overall_rc);
+    execute_ovsdb_command("set interface patch-int type=patch",
+                          not_care_culminative_time, overall_rc);
 
-  execute_ovsdb_command("add-port br-tun patch-int", not_care_culminative_time, overall_rc);
+    execute_ovsdb_command("set interface patch-int options:peer=patch-tun",
+                          not_care_culminative_time, overall_rc);
 
-  execute_ovsdb_command("set interface patch-int type=patch",
-                        not_care_culminative_time, overall_rc);
+    // adding default flows
+    execute_openflow_command("add-flow br-tun \"table=0, priority=1,in_port=\"patch-int\" actions=resubmit(,2)\"",
```

Comment:
What does table 2 do next?

Reply:
table 2 in br-tun is used to send outgoing traffic to table 22. Table 22 will send traffic to the right output tunnel based on the internal vlan id, table 22 is programmed by create_update_neighbor_port in line 228.